cmd/check_tx_lookup.cpp

```diff
@@ -0,0 +1,108 @@
+#include <silkworm/db/stages.hpp>
+#include <silkworm/types/transaction.hpp>
+#include <boost/endian/conversion.hpp>
+#include <silkworm/common/util.hpp>
+#include <silkworm/chain/config.hpp>
+#include <iostream>
+
+using namespace silkworm;
+ 
+int main(int argc, char* argv[]) { 
+    namespace fs = boost::filesystem;
+    
+    CLI::App app{"Check Tx Hashes => BlockNumber mapping in database"};
+
+    std::string db_path{db::default_path()};
+    size_t block_from;
+    app.add_option("-d,--datadir", db_path, "Path to a database populated by Turbo-Geth", true)
+        ->check(CLI::ExistingDirectory);
+    app.add_option("--from", block_from, "Initial block number to process (inclusive)", true)
+        ->check(CLI::Range(1u, UINT32_MAX));
+    CLI11_PARSE(app, argc, argv);
+
+    Logger::default_logger().set_local_timezone(true);  // for compatibility with TG logging
+
+    // Check data.mdb exists in provided directory
+    boost::filesystem::path db_file{boost::filesystem::path(db_path) / boost::filesystem::path("data.mdb")};
+    if (!boost::filesystem::exists(db_file)) {
+        SILKWORM_LOG(LogError) << "Can't find a valid TG data file in " << db_path << std::endl;
+        return -1;
+    }
+    fs::path datadir(db_path);
+    fs::path etl_path(datadir.parent_path() / fs::path("etl-temp"));
+    fs::create_directories(etl_path);
+    etl::Collector collector(etl_path.string().c_str(), /* flush size */ 512 * kMebi);
+
+    lmdb::DatabaseConfig db_config{db_path};
+    db_config.set_readonly(false); 
+    std::shared_ptr<lmdb::Environment> env{lmdb::get_env(db_config)};
+    std::unique_ptr<lmdb::Transaction> txn{env->begin_ro_transaction()};
+
+    auto bodies_table{txn->open(db::table::kBlockBodies)};
+    auto tx_lookup_table{txn->open(db::table::kTxLookup)};
+    auto transactions_table{txn->open(db::table::kEthTx)};
+    uint64_t expected_block_number{0};
+
+    try {
+        Bytes start(8, '\0');
+        boost::endian::store_big_u64(&start[0], 0);
+        MDB_val key_mdb{db::to_mdb_val(start)};
+        MDB_val data_mdb;
+        SILKWORM_LOG(LogInfo) << "Checking Transaction Lookups..." << std::endl;
+        for (int rc{bodies_table->seek(&key_mdb, &data_mdb)}; rc != MDB_NOTFOUND; rc = bodies_table->get_next(&key_mdb, &data_mdb)) {
+            auto body_rlp{db::from_mdb_val(data_mdb)};
+            auto body{db::detail::decode_stored_block_body(body_rlp)};
+            if (body.txn_count == 0) {
+                expected_block_number++;
+                continue;
+            }
+            Bytes transaction_key(8, '\0');
+            boost::endian::store_big_u64(transaction_key.data(), body.base_txn_id);
+            MDB_val tx_key_mdb{db::to_mdb_val(transaction_key)};
+            MDB_val tx_data_mdb;
+
+            uint64_t i{0};
+            for (int rc{transactions_table->seek_exact(&tx_key_mdb, &tx_data_mdb)}; rc != MDB_NOTFOUND && i < body.txn_count; rc = transactions_table->get_next(&tx_key_mdb, &tx_data_mdb), ++i) {
```

Comment:
This loop can finish only if you reach the limit of `body.txn_count` **or** `rc == MDB_NOT_FOUND`
You're ignoring every possible other db error which may cause you to insert wrong data into collector.